sympy/matrices/tests/test_matrices.py

```diff
@@ -2467,3 +2467,28 @@ def test_doit():
     a = Matrix([[Add(x,x, evaluate=False)]])
     assert a[0] != 2*x
     assert a.doit() == Matrix([[2*x]])
+
+def test_issue_9457():
+    # for row_del(index)
+    M = Matrix([[1, 2, 3], [2, 3, 4], [3, 4, 5]])
+    M.row_del(1)
+    assert M == Matrix([[1, 2, 3], [3, 4, 5]])
+    N = Matrix([[1, 2, 3], [2, 3, 4], [3, 4, 5]])
+    N.row_del(-2)
+    assert N == Matrix([[1, 2, 3], [3, 4, 5]])
+    P = Matrix([[1, 2, 3], [2, 3, 4], [3, 4, 5]])
+    raises(ValueError, lambda: P.row_del(10))
```

Comment:
I think `ValueError` is better since i am using code `del self._mat[i:j]` now since the deletion this way of any "out of bound" index is feasible.
For ex.
```
>>> a = [1, 2, 3, 4]
del a[5: 30]
```
this is valid and is performed with no IndexError.
So the point is "I have changed previous code of re-assigning the `self._mat` as was done earlier in row_del, that was inefficient in terms of time". So my preference will be to raise `ValueError` because the value of Row to be deleted is not valid though the internal operation of deletion is valid.